### DISCUSSION TITLE
Shelly1 Pulsetimer interrupt

### DISCUSSION THREAD

**Discussion by rasti70** [arendst/Tasmota#10532] [Q&A] [answered]
Hello,

I use a Shelly1 as Eltako replacement. Several push-buttons can switch the light on and off independently. 

A timer function that automatically switches the light off again after x seconds also works. 

Switchmode1 = 4
Pulsetime1= 0 for simple Eltako function
Pulsetime1= 115 for automatic light off after 15 seconds 

Question: 
How can I interrupt the pulsetimer by pressing the button again? 
What I want to do is to manually switch the light off before the pulsetimer automatically switches the light off .

Thanks in advance for your help

Ralf

**Comment by barbudor** [comment, marked as answer]
Hi

You can interrupt a running Pulsetimer with `power off`
You should be able to achieve what you are looking for with the following rule :
```
rule1 
    on power1#state=1 do 
        var1 0 
    endon 
    on power1#state=0 do 
        var1 1 
    endon 
    on switch1#state do 
        power %var1% 
    endon
```
NOT TESTED

Let us know if it works

**Reply by rasti70** [reply]
works like a charm, thank you so much !
